src/utils/CustomError.ts

```diff
@@ -1,9 +1,10 @@
-export class CustomError<T = string> {
-  public message: T;
+// TODO: create specific errors to avoid generic
```

Comment:
Kind of overhead......
I can't think of any reason why this might be necessary, can you tell me?

Reply:
After moving db logic to services, to encapsulate logic of http responses we can create specific errors, and define error response codes in error layer.
Let's discuss.